### PR TITLE
Update tests action

### PR DESCRIPTION
**Pull request by waketzheng**
1. Update cache action version
2. Remove quotes around 3.11
3. Use map to replace list comprehension

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -34,7 +34,7 @@ jobs:
         id: full-python-version
         shell: bash
         run: |
-          echo ::set-output name=version::$(python -c "import sys; print('-'.join(str(v) for v in sys.version_info))")
```
Comment (frostming) [inline .github/workflows/tests.yml:37]: Can you change this to the new `GITHUB_OUTPUT` syntax?

Comment (waketzheng) [reply .github/workflows/tests.yml:37]: Do you mean: `echo "{name}={value}" >> "$GITHUB_OUTPUT"`?

Comment (frostming) [reply .github/workflows/tests.yml:37]: Right

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -34,7 +34,7 @@ jobs:
...
+          echo name=version::$(python -c "import sys; print('-'.join(map(str, sys.version_info)))") >> "$GITHUB_OUTPUT"
```
Comment (frostming) [inline .github/workflows/tests.yml:37]:
```suggestion
          echo version=$(python -c "import sys; print('-'.join(map(str, sys.version_info)))") >> "$GITHUB_OUTPUT"
```

**Review by frostming — APPROVED** [review]

**Pull request merged**